tests/test_authz_authorizer.py

```diff
@@ -1,130 +1,241 @@
+
+    def test_wrong_jwt_authz_payload_raises_permission_denied(self) -> None:
+        with pytest.raises(PermissionDenied):
+            self._make_mocked_authorizer({})
+
+    def test_check_deny_res(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        authz.deny = {"test_resource": ALL}
+        with pytest.raises(PermissionDenied):
+            authz._check_deny()  # pylint: disable=protected-access
+        assert authz.outcome == DENY
+
+    def test_check_allow_all(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+        assert authz.outcome == ALLOW
+
+    def test_check_deny_any(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        authz.deny = {ALL: ALL}
+        with pytest.raises(PermissionDenied):
+            authz._check_deny()  # pylint: disable=protected-access
+        assert authz.outcome == DENY
+
+    def test_check_access_check_deny_outcome(self) -> None:
+        authz = self._make_mocked_authorizer({"allow": {}, "deny": {ALL: ALL}})
+        with pytest.raises(PermissionDenied):
+            authz.check_access()
+
+    def test_check_access_outcome_deny(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer({**full_access_authz_payload, "allow": {ALL: None}})
+        with pytest.raises(PermissionDenied):
+            authz.check_access()
+
+    def test__check_resource(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        with patch.object(authz, "_deny_if_all", new_callable=MagicMock()) as mocked_deny:
+            authz._check_resource({"res": "xxx"})  # pylint: disable=protected-access
+            mocked_deny.assert_has_calls(
+                [
+                    call({"res": "xxx"}),
+                    call("res"),
+                    call("xxx"),
+                ]
+            )
+
+    def test_validate_allow_all(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        authz.check_access()
+        assert authz.outcome == ALLOW
+        assert authz.restrictions == {"allow": "*", "deny": None}
+
+    def test_check_allow_fail_all(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        authz.allow = None
+        with pytest.raises(PermissionDenied):
+            authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+            assert authz.outcome == ALLOW
+
+    def test_check_allow_resource(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        authz.allow = {"test_resource": ALL}
```

Comment:
Can't you just create an object you need for the test instead of changing it one line below? 🤔